### DISCUSSION TITLE
NextJS Hacked Async Support

### DISCUSSION THREAD

**Discussion by arranf** [markdoc/markdoc#101] [Show and tell]
I've been using Markdoc in my Next JS site to write a "book" but there are some elements I want to pull from an API which unfortunately requires async behaviour.

I also wanted this to happen in the [getStaticProps](https://nextjs.org/docs/basic-features/data-fetching/get-static-props) step to enable search using [Stork Search](https://stork-search.net/).

I [forked the markdoc nextjs plugin](https://github.com/arranf/markdoc-next.js). I use frontmatter to define a set of "loaders" (async functions) which return data that is passed as props to partials and page components.

These can then be accessed by nodes/tags in the `config.variables` object.

This is _far_ from perfect as ideally tags/nodes would be able to influence the data request via attributes rather than me creating a one-size fits all "overfetch" for the page but this does seems to solve my need for now.

**Comment by haikyuu** [comment]
Very cool & interesting, can you share a bit more about the book setup? I'm also planning to make a book using markdoc :)
